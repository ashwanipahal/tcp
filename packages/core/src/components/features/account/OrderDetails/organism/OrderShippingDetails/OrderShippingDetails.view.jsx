import React from 'react';
import PropTypes from 'prop-types';
import { BodyCopy } from '@tcp/core/src/components/common/atoms';
import withStyles from '@tcp/core/src/components/common/hoc/withStyles';
import Address from '@tcp/core/src/components/common/molecules/Address';
import { getLabelValue } from '@tcp/core/src/utils/utils';
import styles from './styles/OrderShippingDetails.style';

/**
 * This function component use for return the OrderShippingDetails
 * can be passed in the component.
 * @param OrdersLabels - OrdersLabels object used for showing Orders Labels
 */

const OrderShippingDetails = ({ className, OrderDetailsData, OrdersLabels }) => {
  /**
   * @function return  Used to render the JSX of the component
   * @param    {[Void]} function does not accept anything.
   * @return   {[Object]} JSX of the component
   */

  const { checkout } = OrderDetailsData;
  const { shippingAddress } = checkout;

  return (
    <BodyCopy component="div" className={className}>
      <BodyCopy
        fontSize="fs14"
        fontWeight="extrabold"
        fontFamily="secondary"
        className="elem-mb-MED"
      >
        {getLabelValue(OrdersLabels, 'lbl_orderDetails_shipping')}
      </BodyCopy>
      <BodyCopy component="div" fontSize="fs14" fontFamily="secondary">
        <Address address={shippingAddress} showCountry={false} showPhone={false} />
      </BodyCopy>
    </BodyCopy>
  );
};
OrderShippingDetails.propTypes = {
  className: PropTypes.string,
  OrderDetailsData: PropTypes.shape([]),
  OrdersLabels: PropTypes.shape({
    lbl_orderDetails_shipping: PropTypes.string,
  }),
};

OrderShippingDetails.defaultProps = {
  className: '',
  OrderDetailsData: [],
  OrdersLabels: {
    lbl_orderDetails_shipping: '',
  },
};

export default withStyles(OrderShippingDetails, styles);
export { OrderShippingDetails as OrderShippingDetailsVanilla };
