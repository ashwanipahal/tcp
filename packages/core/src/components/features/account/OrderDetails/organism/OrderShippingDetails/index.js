export { default } from './views/OrderShippingDetails.view';
