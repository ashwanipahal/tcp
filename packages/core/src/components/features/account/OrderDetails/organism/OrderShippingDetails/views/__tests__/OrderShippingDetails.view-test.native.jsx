import React from 'react';
import { shallow } from 'enzyme';
import OrderShippingDetails from '../OrderShippingDetails.view';

describe('Order Shipping Details component', () => {
  it('should renders correctly with ECOM order', () => {
    const props = {
      orderDetailsData: {
        checkout: {
          shippingAddress: {
            addressLine1: 'Wesley Street ',
            addressLine2: '',
            city: 'Bronx',
            country: 'United States',
            firstName: 'User1',
            lastName: 'UserName1',
            state: 'NY',
            zipCode: '10451',
          },
        },
        ordersLabels: {},
      },
    };
    const component = shallow(<OrderShippingDetails {...props} />);
    expect(component).toMatchSnapshot();
  });
  it('should renders correctly with BOSS/BOPIS order', () => {
    const props = {
      orderDetailsData: {
        checkout: {
          pickUpStore: {
            basicInfo: {
              storeName: '28152 LA drive',
            },
            address: {
              addressLine1: '234 E 149th St',
              city: 'Bronx',
              state: 'NY',
              zipCode: '10451',
            },
          },
          pickUpPrimary: {
            firstName: 'User2',
            lastName: 'User2',
          },
        },
        ordersLabels: {},
      },
    };
    const component = shallow(<OrderShippingDetails {...props} />);
    expect(component).toMatchSnapshot();
  });
});
