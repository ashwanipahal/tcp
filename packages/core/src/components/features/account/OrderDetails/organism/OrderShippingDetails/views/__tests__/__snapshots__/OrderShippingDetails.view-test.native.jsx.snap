// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order Shipping Details component should renders correctly with BOSS/BOPIS order 1`] = `
<Fragment>
  <Styled(View)
    spacingStyles="margin-top-MED"
  >
    <Styled(Styled(BodyCopy))
      accessibilityText=""
      dataLocator=""
      fontFamily="primary"
      fontSize="fs14"
      fontWeight="semibold"
      margin={null}
      numberOfLines={null}
      spacingStyles="margin-top-XS margin-bottom-XS"
      text="lbl_orderDetails_pickup"
      textDecoration={null}
    />
    <Address
      className="elem-mb-SM elem-mt-XXS"
      dataLocatorPrefix="address"
      fontSize="fs14"
      regularName={true}
      showCountry={false}
      showDefaultText={false}
      showName={true}
      showPhone={false}
    />
  </Styled(View)>
  <PickUpSummary
    ordersLabels={Object {}}
    pickUpStore={
      Object {
        "address": Object {
          "addressLine1": "234 E 149th St",
          "city": "Bronx",
          "state": "NY",
          "zipCode": "10451",
        },
        "basicInfo": Object {
          "storeName": "28152 LA drive",
        },
      }
    }
  />
</Fragment>
`;

exports[`Order Shipping Details component should renders correctly with ECOM order 1`] = `
<Fragment>
  <Styled(View)
    spacingStyles="margin-top-MED"
  >
    <Styled(Styled(BodyCopy))
      accessibilityText=""
      dataLocator=""
      fontFamily="primary"
      fontSize="fs14"
      fontWeight="semibold"
      margin={null}
      numberOfLines={null}
      spacingStyles="margin-top-XS margin-bottom-XS"
      text="lbl_orderDetails_shipping"
      textDecoration={null}
    />
    <Address
      address={
        Object {
          "addressLine1": "Wesley Street ",
          "addressLine2": "",
          "city": "Bronx",
          "country": "United States",
          "firstName": "User1",
          "lastName": "UserName1",
          "state": "NY",
          "zipCode": "10451",
        }
      }
      className="elem-mb-SM elem-mt-XXS"
      dataLocatorPrefix="address"
      fontSize="fs14"
      regularName={true}
      showCountry={false}
      showDefaultText={false}
      showName={true}
      showPhone={false}
    />
  </Styled(View)>
</Fragment>
`;
